ansibullbot/utils/extractors.py

```diff
@@ -509,61 +511,52 @@ def get_module_authors(self):
                 break
             if inphase:
                 documentation += line + u'\n'
-
-        if not documentation:
-            logging.debug(u'no documentation found in {}'.format(self.filepath))
-            return []
-
-        # clean out any other yaml besides author to save time
-        inphase = False
-        author_lines = u''
+    
         self._DOCUMENTATION_RAW = documentation
-        doc_lines = documentation.split(u'\n')
-        for idx, x in enumerate(doc_lines):
-            if x.startswith(u'author'):
-                #print("START ON %s" % x)
-                inphase = True
-                #continue
-            if inphase and not x.strip().startswith((u'-', u'author')):
-                #print("BREAK ON %s" % x)
-                inphase = False
-                break
-            if inphase:
-                author_lines += x + u'\n'
 
-        if not author_lines:
-            logging.debug(u'no author lines found in {}'.format(self.filepath))
-            return []
-
-        self._AUTHORS_RAW = author_lines
-        ydata = {}
+        # some docstrings don't pass yaml validation with PyYAML >= 4.2
         try:
-            ydata = yaml.load(author_lines, BotYAMLLoader)
-            self._AUTHORS_DATA = ydata
-        except Exception as e:
-            print(e)
-            return []
+            self._DOCSTRING = yaml.load(self._DOCUMENTATION_RAW)
+        except yaml.parser.ParserError as e:
+            #logging.debug(e)
+            logging.warning('%s has non-yaml formatted docstrings' % self.filepath)
+        except yaml.scanner.ScannerError as e:
+            #logging.debug(e)
+            logging.warning('%s has non-yaml formatted docstrings' % self.filepath)
 
-        # quit early if the yaml was not valid
-        if not ydata:
-            return []
+        # always cast to a dict for easier handling later
+        if self._DOCSTRING is None:
+            self._DOCSTRING = {}
 
-        # quit if the key was not found
-        if u'author' not in ydata:
-            return []
+        return self._DOCSTRING
 
-        if type(ydata[u'author']) != list:
-            ydata[u'author'] = [ydata[u'author']]
+    def get_module_authors(self):
+        """Grep the authors out of the module docstrings"""
 
-        authors = []
-        for author in ydata[u'author']:
-            github_ids = self.extract_github_id(author)
-            if github_ids:
-                authors.extend(github_ids)
+        # 2019-02-15
+        if 'author' in self.docs or 'authors' in self.docs:
```

Comment:
This check could be simplified:
```suggestion
        _authors = self.docs.get('author') or self.docs.get('authors')
        if not _authors:
            return []

        # rest of code is unindented, extra similar check is not needed
```